docs/iris/gallery_code/meteorology/plot_lagged_ensemble.py

```diff
@@ -51,7 +52,7 @@ def main():
     # coordinate, adding appropriate lagged ensemble metadata
     surface_temp = iris.load_cube(
         iris.sample_data_path("GloSea4", "ensemble_???.pp"),
-        iris.Constraint("surface_temperature", realization=lambda value: True),
```

Comment:
I don't actually understand the realization part of this constraint but removing it didn't make any difference to the plot.

Reply:
Wow. Okay, what the heck... hmmm without digging this seems like an ancient workaround for something.

The only thing I can think of is that it's ensuring that the `realization` coordinate is always present? Dunno... anyways if removing it makes no difference, then let's remove it. Great spot :eyes:

Reply:
OK, because I hate not knowing, I had a play:

* If I don't use the callback or the realization constraint, I get a `MergeError` because of mismatched "realization" coordinates, as we might expect.
* If I don't use the callback but reinstate the realization constraint, the plot works fine but more maps  and lines are missing.  So yeah, it seems this is a "coordinate exists" constraint, which actually seems quite useful, though I guess you could get the same effect using a `cube_func`.

Reply:
@rcomer Well at the very least it deserves a comment to explain what the heck it's doing. 

It's a neat pattern to employ without cracking open the `cube_func`. So let's honour the original authors intent but with some clarification for the reader... what say ye? :thinking:

Reply:
Definitely: if we're going to include it, it should be explained!  But... if we advertise this behaviour, should there also be a test for this behaviour?  _Is_ there a test for this behaviour?  If not, should I add one, and if so where?

And I thought a gallery update would be straightforward 🙄 🤣

Reply:
No worries. Let's conquer and divide here :muscle: 

How's about you add a comment, if you're happy doing that and I'll add some appropriate test coverage, and target a pull-request on your `rcomer:gallery-seasonal-ensemble` branch.

Deal? 

(Ode to a simple life :pray:)

Reply:
Yeah sounds good to me.

Reply:
I figured it's easier to explain the constraint if it's defined on its own line.